src/modules/components/address/addressInput/addressInput.tsx

```diff
@@ -0,0 +1,250 @@
+import { useQueryClient } from '@tanstack/react-query';
+import classNames from 'classnames';
+import { forwardRef, useEffect, useRef, useState, type ChangeEvent, type FocusEvent } from 'react';
+import { type Address } from 'viem';
+import { normalize } from 'viem/ens';
+import { useConfig, useEnsAddress, useEnsName, type UseEnsAddressParameters, type UseEnsNameParameters } from 'wagmi';
+import {
+    Button,
+    IconType,
+    InputContainer,
+    Spinner,
+    clipboardUtils,
+    mergeRefs,
+    useDebouncedValue,
+    useInputProps,
+    type IInputComponentProps,
+} from '../../../../core';
+import type { IWeb3ComponentProps } from '../../../types';
+import { addressUtils, ensUtils } from '../../../utils';
+import { MemberAvatar } from '../../member';
+
+export interface IAddressInputResolvedValue {
+    /**
+     * Address value.
+     */
+    address?: string;
+    /**
+     * ENS name linked to the given address.
+     */
+    name?: string;
+}
+
+export interface IAddressInputProps
+    extends Omit<IInputComponentProps<HTMLTextAreaElement>, 'maxLength' | 'value' | 'onChange'>,
+        IWeb3ComponentProps {
+    /**
+     * Current value of the address input.
+     */
+    value?: string;
```

Comment:
Yes, currently if the parent component already has the name/address pair, the address-input will still make the resolving / RPC requests. In general, if we always use the `useEnsAddress` / `useEnsName` hooks from Wagmi on the app, the cache would be prepopulated and the component will just use the react-query cache. If we need, we could update the component to support a new parameter (like `initialData` ?) to use to prepopulate the react-query / wagmi cache

Reply:
The use case I'm bringing this up for is if this data is being returned pre-resolved by a backend for example.  In that case there's no prepopulated cache for this component to read. Because value is a single string prop, it would be nice to be able to use something like `initialData` for this use case. Can you add an improvement ticket for that?

Reply:
Yes makes sense, created a ticket here => https://aragonassociation.atlassian.net/browse/APP-2998